### PR TITLE
Dry up TransportResponseHandler use via ActionListenerResponseHandler

### PR DESCRIPTION
**Pull request by original-brownbear**
Use the action listener response handler in a couple additional spots to get proper `toString` that make slow logging easier to interpret if needed and to save some LoC and duplication.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/ActionListenerResponseHandler.java [hunk 1]
```diff
@@ -24,7 +24,7 @@
  */
 public class ActionListenerResponseHandler<Response extends TransportResponse> implements TransportResponseHandler<Response> {
 
-    private final ActionListener<? super Response> listener;
+    protected final ActionListener<? super Response> listener;
```
Comment (idegtiarenko) [inline server/src/main/java/org/elasticsearch/action/ActionListenerResponseHandler.java:27]: Is this change required? It looks like all of the updated usages are referencing same listener passed in constructor or am I missing something?

Comment (original-brownbear) [reply server/src/main/java/org/elasticsearch/action/ActionListenerResponseHandler.java:27]: This way we save a reference in some, because we don't have to capture `listener` twice (once in the variable and once in the anonymous capture). We have the same approach in `ActionRunnable` so I did it here also to save some references (if only to make heap dumps easier to grind through :D).

**Review by idegtiarenko — APPROVED** [review]
Left a comment, otherwise LGTM

**Comment by original-brownbear** [comment]
Thanks Ievgen!
